src/main/kotlin/ro/hibyte/betting/service/PrizeDrawService.kt

```diff
@@ -88,6 +123,8 @@ class PrizeDrawService(
         verifyEntryBadRequest(foundPrizeDraw, userProfile, prizeDrawEntryDTO)
         if(foundPrizeDraw.entries.any { it.user.userId == userProfile.userId }) throw ConflictException("User already has an entry in this draw")
 
+        userProfile.coins = userProfile.coins.toInt() - prizeDrawEntryDTO.amount.toInt()
```

Comment:
I am not sure I understand the type that the "coins" field has. Sometimes it is used as Int sometimes as Double. 

One idea is that the coins field in the User is Double (since winnings from a bet will be "doubles"), but amounts from Bet should be Integers, since I will only be able to bet an integer amount of coins in a bet.

Another way, is to assume that both coins and amount are integers and to compute the winnings of a bet by always rounding up (to the users advantage) to the nearest integer

Reply:
Anyway, please change the types of the coins, and of the amount in bot prizes and bet. I would not advise to use Number as type.

Reply:
however I understand that this is not part of this PR